src/triangular.jl

```diff
@@ -0,0 +1,63 @@
+struct Triangular{B <: NTuple{2, AbstractBoundary}} <: AbstractLattice
+    dims::NTuple{2, Int}
+    bcs::NTuple{2, AbstractBoundary}
+    translation_vectors::Dict{Int, Vector{Coordinate{2, Int}}}
+    function Triangular{B}(dims::NTuple{2, Int}, bcs::B) where B <: NTuple{2, AbstractBoundary}
+        check_boundaries(bcs)
+        new{B}(dims, bcs, Dict{Int, Vector{Coordinate{2, Int}}}())
+    end
+end
+
+
+for L in [:Honeycomb, :Kagome]
+
+    @eval struct $L{B <: NTuple{2, AbstractBoundary}} <: AbstractLattice
+        dims::NTuple{2, Int}
+        bcs::NTuple{2, AbstractBoundary}
+        translation_vectors::Dict{Tuple{Int, Int}, Vector{Coordinate{3, Int}}}
```

Comment:
if we end up in using translation vectors to do it, should we actually have a type for this then wrap that type to be `Tri` , `Kagome` etc.?

Reply:
i dont understand, could you elaborate?

Reply:
e.g should we have a container lattice type that looks like

```julia
struct TranslationLattice{N, B <: NTuple{N, AbstractBoundary}} <: AbstractLattice
   dims::NTuple{N, Int}
   bcs::B
   translation_vectors
end
```

?